modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/AbstractFilterToMongo.java

```diff
@@ -520,27 +543,88 @@ public Object visit(Disjoint filter, Object extraData) {
-                        .add("$geometry", geometryDBObject)
-                        .add("$maxDistance", distanceInMeters)
-                        .get();
+        CoordinateReferenceSystem coordinateReferenceSystem =
+                featureType.getCoordinateReferenceSystem();
+        double distanceInMeters = DistanceBufferUtil.getDistanceInMeters(filter);
+        DBObject dbo;
+        // If point use $near
+        if (geometry instanceof Point) {
+            DBObject geometryDBObject = geometryBuilder.toObject(geometry);
+            dbo =
+                    BasicDBObjectBuilder.start()
+                            .push("$near")
+                            .add("$geometry", geometryDBObject)
+                            .add("$maxDistance", distanceInMeters)
+                            .get();
+            // Other type of geometry, use $geoIntersects
+        } else {
+            MathTransform transform = null;
+            try {
+                CoordinateReferenceSystem azimuthalEquidistantCRS =
+                        computeAzimutalEquidistantCRS(geometry, coordinateReferenceSystem);
+                // Transform geometry to CRS that supports meters
+                transform =
+                        CRS.findMathTransform(coordinateReferenceSystem, azimuthalEquidistantCRS);
+                geometry = JTS.transform(geometry, transform);
+
+                // Apply buffer in meters to geometry
+                Geometry bufferedGeometry = geometry.buffer(distanceInMeters, DEFAULT_SEGMENTS);
+
+                // Transform back to original with buffer applied
+                transform =
+                        CRS.findMathTransform(azimuthalEquidistantCRS, coordinateReferenceSystem);
+                geometry = JTS.transform(bufferedGeometry, transform);
+            } catch (FactoryException | TransformException e) {
+                throw new RuntimeException(e);
+            }
+            DBObject geometryDBObject = geometryBuilder.toObject(geometry);
+            dbo =
+                    BasicDBObjectBuilder.start()
+                            .push("$geoIntersects")
+                            .add("$geometry", geometryDBObject)
+                            .get();
+        }
 
         output.put((String) e1, dbo);
         return output;
     }
 
+    private static CoordinateReferenceSystem computeAzimutalEquidistantCRS(
+            Geometry geometry, CoordinateReferenceSystem coordinateReferenceSystem)
+            throws FactoryException, TransformException {
+        Point centroid = geometry.getCentroid();
+        if (!(coordinateReferenceSystem instanceof GeographicCRS)) {
+            MathTransform mathTransform =
+                    CRS.findMathTransform(coordinateReferenceSystem, DefaultGeographicCRS.WGS84);
+            centroid = (Point) JTS.transform(centroid, mathTransform);
+        }
+
+        double lon = centroid.getX();
+        double lat = centroid.getY();
```

Comment:
Mind that a GeographicCRS can still be in NORTH/EAST axis order (where x is latitude). One easy way out is to add a condition to the if above, testing CRS.getAxisOrder(coordinateReferenceSsytem)... the reprojection to WGS84 will also give you the axis order you want.

Reply:
Added check to if condition mentioned above